buildSrc/build.gradle.kts

```diff
@@ -148,7 +152,38 @@ if (!isCiServer || System.getProperty("enableCodeQuality")?.toLowerCase() == "tr
     apply { from("../gradle/codeQuality.gradle") }
 }
 
-apply { from("../gradle/ciReporting.gradle") }
+if (isCiServer) {
+    gradle.buildFinished {
+        allprojects {
+            tasks.all {
+                if (this is Reporting<*> && state.failure != null) {
+                    val reportContainer = this.reports
+                    val reportDestination = reportContainer.getByName("html").destination
+                    prepareReportForCIPublishing(this@allprojects, reportDestination, this@allprojects.name)
+                }
+            }
+        }
+    }
+}
+
+fun prepareReportForCIPublishing(project: Project, report: File, projectName: String) {
```

Comment:
:x: I'd also make this function a `Project` extension to avoid one level of nesting.

Reply:
Done.